e2e/jenkins/crw-ci/pr-check/k8s/Jenkinsfile

```diff
@@ -14,225 +14,294 @@ pipeline {
+                        ((SUCCESS_RATE_COUNTER+=1))
+                      fi
+                      sleep 1;
+                    
+                      if [ \$SUCCESS_RATE_COUNTER == \$SUCCESS_THRESHOLD ]; then 
+                        echo "Route is stable enough. Continuing running tests"
+                        break
+                      fi
+                     done
+                """
             }
         }
 
         stage("Create test workspace") {
+            when {
+                expression { return testWorkspaceDevfileUrl != null && testWorkspaceDevfileUrl != "" }
+            }
+
             steps {
-                script {
-                    sh "${WORKSPACE}/chectl workspace:start --devfile=$DEVFILE_URL"
-                }
+                sh """
+                    wget $testWorkspaceDevfileUrl -O ${WORKSPACE}/test-workspace-devfile.yaml
+
+                    KEYCLOAK_URL=\$(kubectl get ingress/keycloak -n che -o jsonpath='{.spec.rules[0].host}')
+                    KEYCLOAK_BASE_URL="http://\${KEYCLOAK_URL}/auth"
+
+                    USER_ACCESS_TOKEN=\$(curl -X POST \$KEYCLOAK_BASE_URL/realms/che/protocol/openid-connect/token \\
+                       -H "Content-Type: application/x-www-form-urlencoded" \\
+                       -d "username=admin" \\
+                       -d "password=admin" \\
+                       -d "grant_type=password" \\
+                       -d "client_id=che-public" | jq -r .access_token)
+
+                    ${WORKSPACE}/chectl workspace:start \\
+                       --devfile=${WORKSPACE}/test-workspace-devfile.yaml \\
+                       --access-token "\$USER_ACCESS_TOKEN"
+                """
             }
         }
 
         stage("Run E2E Happy path tests") {
             steps {
-                script {
-                    sh """
-                         docker pull eclipse/che-e2e:nightly
-                         
-                         CHE_HOST=\$(kubectl get ingress che-ingress -n=che -o=jsonpath={'.spec.rules[0].host'})
-                         CHE_URL=http://\${CHE_HOST}
-                         docker run --shm-size=256m --net=host --ipc=host \\
-                           -e TS_SELENIUM_HEADLESS='true' \\
-                           -e TS_SELENIUM_DEFAULT_TIMEOUT=300000 \\
-                           -e TS_SELENIUM_LOAD_PAGE_TIMEOUT=240000 \\
-                           -e TS_SELENIUM_WORKSPACE_STATUS_POLLING=20000 \\
-                           -e TS_SELENIUM_BASE_URL=\${CHE_URL} \\
-                           -e TS_SELENIUM_LOG_LEVEL='DEBUG' \\
-                           -v ${WORKSPACE}/e2e:/tmp/e2e:Z \\
-                           eclipse/che-e2e:nightly
-                    """
-                }
+                sh """
+                     docker pull eclipse/che-e2e:nightly
+                     
+                     CHE_HOST=\$(kubectl get ingress che -n=che -o=jsonpath={'.spec.rules[0].host'})
```

Comment:
CHE_HOST is obtained already on line 169. Can it be reused?

Reply:
It's a good point. Thanks.